src/pybind/mgr/cephadm/agent.py

```diff
@@ -44,212 +42,89 @@ def cherrypy_filter(record: logging.LogRecord) -> int:
-                if service['type'] != 'mgr':
-                    continue
-                port = self.mgr.get_module_option_ex('prometheus', 'server_port', 9283)
-                targets.append(f'{hostname}:{port}')
-        return [{"targets": targets, "labels": {}}]
-
-    def alertmgr_sd_config(self) -> List[Dict[str, Collection[str]]]:
-        """Return <http_sd_config> compatible prometheus config for mgr alertmanager service."""
-        srv_entries = []
-        for dd in self.mgr.cache.get_daemons_by_service('alertmanager'):
-            assert dd.hostname is not None
-            addr = dd.ip if dd.ip else self.mgr.inventory.get_addr(dd.hostname)
-            port = dd.ports[0] if dd.ports else 9093
-            srv_entries.append('{}'.format(build_url(host=addr, port=port).lstrip('/')))
-        return [{"targets": srv_entries, "labels": {}}]
-
-    def node_exporter_sd_config(self) -> List[Dict[str, Collection[str]]]:
-        """Return <http_sd_config> compatible prometheus config for node-exporter service."""
-        srv_entries = []
-        for dd in self.mgr.cache.get_daemons_by_service('node-exporter'):
-            assert dd.hostname is not None
-            addr = dd.ip if dd.ip else self.mgr.inventory.get_addr(dd.hostname)
-            port = dd.ports[0] if dd.ports else 9100
-            srv_entries.append({
-                'targets': [build_url(host=addr, port=port).lstrip('/')],
-                'labels': {'instance': dd.hostname}
-            })
-        return srv_entries
-
-    def haproxy_sd_config(self) -> List[Dict[str, Collection[str]]]:
-        """Return <http_sd_config> compatible prometheus config for haproxy service."""
-        srv_entries = []
-        for dd in self.mgr.cache.get_daemons_by_type('ingress'):
-            if dd.service_name() in self.mgr.spec_store:
-                spec = cast(IngressSpec, self.mgr.spec_store[dd.service_name()].spec)
-                assert dd.hostname is not None
-                if dd.daemon_type == 'haproxy':
-                    addr = self.mgr.inventory.get_addr(dd.hostname)
-                    srv_entries.append({
-                        'targets': [f"{build_url(host=addr, port=spec.monitor_port).lstrip('/')}"],
-                        'labels': {'instance': dd.service_name()}
-                    })
-        return srv_entries
-
-    @cherrypy.expose(alias='prometheus/rules')
-    def get_prometheus_rules(self) -> str:
-        """Return currently configured prometheus rules as Yaml."""
-        cherrypy.response.headers['Content-Type'] = 'text/plain'
-        with open(self.mgr.prometheus_alerts_path, 'r', encoding='utf-8') as f:
-            return f.read()
-
-
-class HostData:
+class HostData(Server):
     exposed = True
 
-    def __init__(self, mgr: "CephadmOrchestrator"):
+    def __init__(self, mgr: "CephadmOrchestrator", port: int, host: str, ssl_ca_cert: str, ssl_priv_key: str):
         self.mgr = mgr
+        Server.__init__(self)
```

Comment:
Did I miss some reply or something? I remember I also pointed out this one.
```suggestion
        super().__init__()
```

Reply:
I think you commented this on the other PR (the one that enables security) as it shares a lot of code with this one. Actually it's based on this PR.

Reply:
ahh, that could be it! 🙈